### ISSUE TITLE
Randomly pick one color for each event written to file (jamp2 handling)

### ISSUE THREAD

**Issue by valassi** [madgraph5/madgraph4gpu#402]
Randomly pick one color for each event written to file (jamp2 handling)

Will give details elsewhere

**Comment by valassi** [comment]
Full details are given in #404.

This is on the critical path to get unweighted event sets to the experiments.

**Comment by oliviermattelaer** [comment]
note that https://github.com/oliviermattelaer/mg5amc_test/issues/24
is changing the fortran handling of color/helicity in order to have a clean api 
This means:
- the fortran/cpp api is now changed to the what we discussed a while ago (input: 2 random number, output: 2 integer)
- New fortran function has been added to the code
  - select_color(rcol, jamp2, channel, proc_id,  icol [out]) 
    - [ ] need to write such function within the CPP plugin (work should have started in that direction before the break)
  - get_helicities(..)
    - no need to convert such function to C++

**Comment by valassi** [comment]
This is in progress. I have some agorithm but it gives me different results from Fortran and C++ (the last two columns).

Maybe not ideal to use gg_tt for debugging as it has only two colors...
```
  33  0.66637210E-01  0.66637210E-01          1.00000000000  3  3   1   1
  34  0.36652604E-01  0.36652604E-01          1.00000000000  8  8   1   1
  35  0.41960814E-01  0.41960814E-01          1.00000000000 12 12   1   1
  36  0.64245071E-01  0.64245071E-01          1.00000000000  2  2   1   1
  37  0.67260622E-01  0.67260622E-01          1.00000000000 15 15   1   1
  38  0.72644897E-01  0.72644897E-01          1.00000000000 15 15   1   1
  39  0.48924468E-01  0.48924468E-01          1.00000000000 15 15   1   1
  40  0.71085828E-01  0.71085828E-01          1.00000000000  2  2   1   1
  41  0.71529802E-01  0.71529802E-01          1.00000000000 15 15   1   1
  42  0.70385722E-01  0.70385722E-01          1.00000000000  2  2   1   1
  43  0.31575040E-01  0.31575040E-01          1.00000000000 12 12   1   1
  44  0.69272519E-01  0.69272519E-01          1.00000000000 15 15   2   1
  45  0.71009335E-01  0.71009335E-01          1.00000000000 15 15   1   1
  46  0.68831440E-01  0.68831440E-01          1.00000000000  2  2   1   1
  47  0.69539468E-01  0.69539468E-01          1.00000000000  2  2   1   1
  48  0.70394921E-01  0.70394921E-01          1.00000000000 15 15   1   1
  49  0.73183046E-01  0.73183046E-01          1.00000000000  3  3   1   1
  50  0.60352285E-01  0.60352285E-01          1.00000000000 15 15   1   1
  51  0.18336225E-01  0.18336225E-01          1.00000000000 15 15   1   1
  52  0.47909190E-01  0.47909190E-01          1.00000000000  2  2   1   1
  53  0.12439986E-01  0.12439986E-01          1.00000000000 15 15   1   1
  54  0.66421684E-01  0.66421684E-01          1.00000000000 15 15   1   1
  55  0.68700645E-01  0.68700645E-01          1.00000000000  2  2   1   1
  56  0.66674348E-01  0.66674348E-01          1.00000000000 12 12   1   1
  57  0.34727484E-01  0.34727484E-01          1.00000000000 15 15   1   2
  58  0.71093852E-01  0.71093852E-01          1.00000000000  2  2   1   1
  59  0.71165949E-01  0.71165949E-01          1.00000000000  2  2   1   1
  60  0.70806691E-01  0.70806691E-01          1.00000000000 15 15   1   1
  61  0.55143515E-01  0.55143515E-01          1.00000000000 15 15   1   1
  62  0.59417397E-01  0.59417397E-01          1.00000000000 15 15   1   1
  63  0.42014188E-01  0.42014188E-01          1.00000000000  5  5   1   1
  64  0.70783989E-01  0.70783989E-01          1.00000000000 15 15   2   2

```

**Comment by valassi** [comment]
There was a silly bug (jamp2_sv not initialised at every SIMD page) in C++. Actually CUDA was already working.

Now everything looks good. This is again from BothDebug, comparing MEs, hel and col:
```
   1  0.64067963E-01  0.64067963E-01          1.00000000000 12 12   1   1
   2  0.58379673E-01  0.58379673E-01          1.00000000000  9  9   1   1
   3  0.70810768E-01  0.70810768E-01          1.00000000000 15 15   1   1
   4  0.67192668E-01  0.67192668E-01          1.00000000000 15 15   1   1
   5  0.71590585E-01  0.71590585E-01          1.00000000000 15 15   1   1
   6  0.72862110E-01  0.72862110E-01          1.00000000000  2  2   1   1
   7  0.14271254E-01  0.14271254E-01          1.00000000000  5  5   1   1
   8  0.63986754E-01  0.63986754E-01          1.00000000000 14 14   2   2
   9  0.46316382E-01  0.46316382E-01          1.00000000000 12 12   1   1
  10  0.35372741E-01  0.35372741E-01          1.00000000000  2  2   1   1
  11  0.73958407E-01  0.73958407E-01          1.00000000000  5  5   1   1
  12  0.70691203E-01  0.70691203E-01          1.00000000000 15 15   1   1
  13  0.70805000E-01  0.70805000E-01          1.00000000000  2  2   1   1
  14  0.30801404E-01  0.30801404E-01          1.00000000000  2  2   1   1
  15  0.64111868E-01  0.64111868E-01          1.00000000000 10 10   2   2
  16  0.74312047E-01  0.74312047E-01          1.00000000000  2  2   1   1
  17  0.60961835E-01  0.60961835E-01          1.00000000000  2  2   1   1
  18  0.67698020E-01  0.67698020E-01          1.00000000000  2  2   1   1
  19  0.49748773E-01  0.49748773E-01          1.00000000000  2  2   1   1
  20  0.71951996E-01  0.71951996E-01          1.00000000000  2  2   1   1
  21  0.52116331E-01  0.52116331E-01          1.00000000000  5  5   1   1
  22  0.69245648E-01  0.69245648E-01          1.00000000000 15 15   1   1
  23  0.64808141E-01  0.64808141E-01          1.00000000000 15 15   1   1
  24  0.66861231E-01  0.66861231E-01          1.00000000000  3  3   2   2
  25  0.70041112E-01  0.70041112E-01          1.00000000000  2  2   1   1
  26  0.61135249E-01  0.61135249E-01          1.00000000000 15 15   1   1
  27  0.66574932E-01  0.66574932E-01          1.00000000000 15 15   1   1
  28  0.67312068E-01  0.67312068E-01          1.00000000000  3  3   1   1
  29  0.47056643E-01  0.47056643E-01          1.00000000000  5  5   2   2
  30  0.70509435E-01  0.70509435E-01          1.00000000000  2  2   2   2
  31  0.23138767E-01  0.23138767E-01          1.00000000000 15 15   1   1
  32  0.76096234E-01  0.76096234E-01          1.00000000000 15 15   1   1
```

This is implemented in MR #573 (currently WIP). When that is merged, this issue will be closed.
